### PR TITLE
[Mellanox] Change NVIDIA SDK build to cmake method

### PR DESCRIPTION
**Pull request by dprital**
#### Why I did it

Up to now, building Mellanox SDK debian packages was done by "autotools" method.
Lately, Mellanox SDK team added the support to build by "cmake" method.
The benefit of using "cmake" is with decrease time of SDK build as "cmake" make it faster then "auto build".

When moving to "cmake" method, all SDK debians:

* applibs
* sx-complib
* sx-example
* sx-gen-utils
* sx-scew
* sxd-libs
* python-sdk-api
* sx-acl-helper
* wjh-libs
* sx-hash-calc
* sx-obj-desc-lib
* sxd-libs

will be replaced by single debian - sys_sdk.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

* Update cmake version on sonic slave dockers (bookworm and bullseye)
* Change SDK Makefiles, delete all Makefiles of removed debians and add new Makefile for building sys_sdk.
* Change sdk.mk and sdk.dep acordingly.
* sx-kernel package was remain since it is consumed by the host as well and I didn't want to consume sys_sdk in host.

#### How to verify it

* Enter syncd docker and verify the list of SDK packages are sx-kernel and sys-sdk only
* Run regression tests and verify there is no degradation
* Install the debian that include dbg-symbols and verify there are debug symbols by setting relevant breakpoints
* Test the build time when using "cmake" for building SDK packages, compare to "auto build"

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: platform/mellanox/mlnx-sai.mk [hunk 1]
```diff
@@ -19,7 +19,7 @@ export MLNX_SAI_VERSION MLNX_SAI_SOURCE_BASE_URL
 
 MLNX_SAI = mlnx-sai_1.mlnx.$(MLNX_SAI_VERSION)_$(CONFIGURED_ARCH).deb
 $(MLNX_SAI)_SRC_PATH = $(PLATFORM_PATH)/mlnx-sai
-$(MLNX_SAI)_DEPENDS += $(MLNX_SDK_DEBS) $(LIBNL_ROUTE3_DEV)
+$(MLNX_SAI)_DEPENDS += $(MLNX_SDK_DEBS) $(LIBNL3_DEV) $(LIBNL_GENL3_DEV) $(LIBNL_ROUTE3_DEV) $(LIBNL_NF3_DEV)
```
Comment (stepanblyschak) [inline platform/mellanox/mlnx-sai.mk:22]: Why now new dependency needed?

File: platform/mellanox/sdk-src/sys-sdk/Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	
+	@echo  'wget -c $(MLNX_SDK_SOURCE_BASE_URL)/$(PACKAGE_NAME)-$(MLNX_SDK_VERSION)-$(MLNX_SDK_ISSU_VERSION).tar.gz'
+	
+	wget -c $(MLNX_SDK_SOURCE_BASE_URL)/$(PACKAGE_NAME)-$(MLNX_SDK_VERSION)-$(MLNX_SDK_ISSU_VERSION).tar.gz -O - | tar -xz
+
+        # build
+
+	rm -rf builds/cmake
+
+	cmake -B builds/cmake -DPYTHON_INTERPRETERS=python3 -DSKIP_UNINSTALL_INSTALL_DEPENDENCY=on -DSKIP_DEPMOD_IN_INSTALL_PHASE=on -DUSE_API_TESTER=OFF -DUSE_KERNEL=OFF -DCMAKE_PROJECT_VERSION=$(MLNX_SDK_DEB_VERSION)
```
Comment (stepanblyschak) [inline platform/mellanox/sdk-src/sys-sdk/Makefile:22]: Does it account for SONIC_DEBUGGING_ON/SONIC_PROFILING_ON?

File: sonic-slave-bookworm/Dockerfile.j2 [hunk 3]
```diff
@@ -420,6 +420,13 @@ RUN apt-get update && apt-get install -y \
         protobuf-compiler \
         libprotobuf-dev
 
+# Upgrade CMAKE version
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.27.6/cmake-3.27.6-linux-x86_64.sh
+RUN chmod +x cmake-3.27.6-linux-x86_64.sh
+RUN sudo ./cmake-3.27.6-linux-x86_64.sh --skip-license --prefix=/usr
```
[the remark this answers is not in the document]
Comment (dprital) [reply sonic-slave-bookworm/Dockerfile.j2:426]: We need the new version of 3.27.6 as the older one doesn't work well

[the remark this answers is not in the document]
Comment (stepanblyschak) [reply sonic-slave-bookworm/Dockerfile.j2:426]: But do we need to install cmake from apt-get just to overwrite it with newer version later on?

File: platform/mellanox/mlnx-sai.mk (hunk 1 above)
Comment (dprital) [reply platform/mellanox/mlnx-sai.mk:22]: Verified that this is redundant so it was removed

File: platform/mellanox/sdk-src/sys-sdk/Makefile [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.ONESHELL:
+SHELL = /bin/bash
+
+MAIN_TARGET = sys-sdk_1.mlnx.$(MLNX_SDK_DEB_VERSION)_$(CONFIGURED_ARCH).deb
+DERIVED_TARGETS = sys-sdk_1.mlnx.$(MLNX_SDK_DEB_VERSION)_$(CONFIGURED_ARCH)-dev.deb sys-sdk_1.mlnx.$(MLNX_SDK_DEB_VERSION)_$(CONFIGURED_ARCH)-dbgsym.ddeb
+
+PACKAGE_NAME = sys_sdk
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+
```
[the remark this answers is not in the document]
Comment (dprital) [reply platform/mellanox/sdk-src/sys-sdk/Makefile:10]: Was removed

File: platform/mellanox/sdk-src/sys-sdk/Makefile [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        # get sources
```
[the remark this answers is not in the document]
Comment (dprital) [reply platform/mellanox/sdk-src/sys-sdk/Makefile:11]: It is at the same place as previous SDK make files. Right above the treatment to get the sources so I odn't see any issue with it.

**Review by stepanblyschak — APPROVED** [review]

File: sonic-slave-bookworm/Dockerfile.j2 [hunk 6]
```diff
@@ -449,7 +449,8 @@ RUN apt-get update && apt-get install -y eatmydata && eatmydata apt-get install
         meson \
         python3-pyelftools \
         hwdata \
-        libzip-dev
+        libzip-dev \
+	libnl-genl-3-dev
```
Comment (stepanblyschak) [inline sonic-slave-bookworm/Dockerfile.j2:453]: Use spaces as in lines above

File: sonic-slave-bullseye/Dockerfile.j2 [hunk 7]
```diff
@@ -425,7 +425,8 @@ RUN apt-get update && apt-get install -y eatmydata && eatmydata apt-get install
         xmlto \
         rake-compiler \
         default-jdk \
-        libgoogle-gson-java
+        libgoogle-gson-java \
+	libnl-genl-3-dev
```
Comment (stepanblyschak) [inline sonic-slave-bullseye/Dockerfile.j2:429]: Same

File: sonic-slave-bookworm/Dockerfile.j2 (hunk 6 above)
Comment (stepanblyschak) [inline sonic-slave-bookworm/Dockerfile.j2:453]: Why need to install it? We build libnl from sources (including libnl-genl-3-dev) - https://github.com/sonic-net/sonic-buildimage/blob/master/rules/libnl3.mk#L20

Comment (dprital) [reply sonic-slave-bookworm/Dockerfile.j2:453]: It is required for cmake build. If I run it without it, it fail

File: sonic-slave-bullseye/Dockerfile.j2 (hunk 7 above)
Comment (dprital) [reply sonic-slave-bullseye/Dockerfile.j2]: Fixed

File: sonic-slave-bookworm/Dockerfile.j2 (hunk 6 above)
Comment (dprital) [reply sonic-slave-bookworm/Dockerfile.j2]: Fixed

Comment (stepanblyschak) [reply sonic-slave-bookworm/Dockerfile.j2]: @dprital The question is why it fails, if libnl-genl-3-dev is built by SONiC there's no need to install it from debian repo. I think there's a dependency on libnl missing in our sdk makefiles

**Review by stepanblyschak — APPROVED** [review]

**Pull request closed without merging**